app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -104,13 +106,27 @@ public void onAttach(Context context) {
     }
 
     @Override
-    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+    public void onCreateOptionsMenu(final Menu menu, MenuInflater inflater) {
         switch (settingsScreen) {
             case SEARCH_ENGINES:
                 inflater.inflate(R.menu.menu_search_engines, menu);
                 break;
             case REMOVE_ENGINES:
                 inflater.inflate(R.menu.menu_remove_search_engines, menu);
+
+                // We disable the trash icon when no engine is checked, enable it otherwise
+                getView().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        MenuItem menuDeleteItem = menu.findItem(R.id.menu_delete_items);
+                        menuDeleteItem.getIcon().mutate().setAlpha(SettingsFragment.ALPHA_GREY);
+                        final MultiselectSearchEngineListPreference pref = (MultiselectSearchEngineListPreference) getPreferenceScreen()
+                                .findPreference(getResources()
+                                        .getString(R.string.pref_key_multiselect_search_engine_list));
+
+                        pref.bindEnginesCheckboxesTo(menuDeleteItem, SettingsFragment.ALPHA_DEFAULT, SettingsFragment.ALPHA_GREY);
```

Comment:
This works and we're pressed for time, so I'd take it but I'll defer to liuche.

That being said, this breaks best practices on view hierarchies: views shouldn't know about their parent views implementation. In this case, we're telling the child view (`MultiselectSearchEngineListPreference`) about `menuDeleteItem` from its parent view. Instead, best practices would dictate:
1. We add a high-level listener to the child view (`onCheckboxChecked`) and the the child view notifies the parent when an event occurs
1. (less ideal but less boilerplate) The parent reaches into the child implementation and adds its own listeners (this is essentially what I'm trying to stop with this comment, but in the other direction)

Why does this matter? The more dependencies there are between different components of the code, the more difficult it becomes to change it. For example:
- If we go with 1), when we change a view, we largely don't have to consider how any views interact with us - we're a black box!
- If 2), you have to consider how the parent interacts with this view's state (one dependency, in one direction down the tree)
- If you let the child know about the parent and you're changing the parent, you have to consider the above and how its children might be touching the parent's internals (two dependencies, but in both directions on the view tree!)

Reply:
These are good tips! In general, I agree with this - but sometimes, the APIs don't work as one would expect. In those situations, it's also helpful to include a short comment explaining something like "colorStateLists don't work with vector drawables" (if that is in fact the case!) so other people reading the code will know why the more complicated, indirect version was used.